test/javascript/testTokenAwareness.js

```diff
@@ -41,11 +41,10 @@ contract('MultiSigWallet', (accounts) => {
         tokenInstance2 = await deployToken()
         const transferAnotherToken = tokenInstance2.contract.transfer.getData(accounts[3], deposit)
 
-        try {
-            await multisigInstance.submitTransaction(tokenInstance2.address, 0, transferAnotherToken, {from: accounts[0]})
-        } catch (e) {
-            // Need better test design for this case but wrapper just awfully fails
-        }
+        multisigInstance.submitTransaction.sendTransaction(tokenInstance2.address, 0, transferAnotherToken, {from: accounts[0]})
```

Comment:
i'm not sure you understand what the promise code is doing here.  may be easier to go over this together on friday.  

yes, it may work fine, but in general, i would stick to either using `await` or using promise chains (`.then()` and `.catch()`).  using both together is kind of messy.

Reply:
I tried to function test () { await testFunction } — it works only for web3.eth.sendTransaction, does not work for multisigInstance.submitTransaction.sendTransaction. I think it has to do something with truffle.

Do you want to switch to full promise chains then?

Reply:
well, if you want the code to wait until line 44 is done (including the catch), then you must put `await` before `multisigInstance` - just adding that one keyword will make the code more consistent.

Reply:
Yes, but we'll still have 2 approaches mixing.

Reply:
yeah, i'm not a real fan of the mixing styles.  but i also appreciate truffle bugs.  

did this code not work (it should be how to do a catch with `await`):

```
try {

  await multisigInstance.submitTransaction
    .sendTransaction(tokenInstance2.address, 0, transferAnotherToken, {from: accounts[0]})
} catch (error) {
  assert.equal(error.message, 'VM Exception while processing transaction: invalid opcode');
}
```